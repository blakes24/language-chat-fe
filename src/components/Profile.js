import ImageUploader from "./ImageUploader";
import {
  Container,
  Typography,
  Grid,
  makeStyles,
  LinearProgress,
  IconButton
} from "@material-ui/core";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Switch from "@material-ui/core/Switch";
import EditIcon from "@material-ui/icons/Edit";
import UserContext from "../helpers/UserContext";
import { useContext, useState } from "react";

const useStyles = makeStyles((theme) => ({
  root: {
    marginTop: "1rem",
    maxWidth: 900,
    [theme.breakpoints.up("sm")]: {
      marginTop: "2rem",
    },
  },
  item1: {
    display: "flex",
    padding: "1rem",
    marginTop: "1rem",
    justifyContent: "center",
    order: 2,
    [theme.breakpoints.up("sm")]: {
      order: 1,
      marginTop: "2rem",
    },
  },
  item2: {
    display: "flex",
    padding: "1rem",
    marginTop: "1rem",
    justifyContent: "center",
    order: 1,
    [theme.breakpoints.up("sm")]: {
      order: 2,
      marginTop: "2rem",
    },
  },
  level: {
    width: "70%",
    marginTop: ".4rem",
    marginBottom: ".4rem",
  },
  switch: {
    marginLeft: ".5rem",
  },
}));

function Profile() {
  const {  user  } = useContext(UserContext);
  const classes = useStyles();
  const [active, setActive] = useState(user.active);

  const handleChange = () => {
    setActive(!active);
  };

  return (
    <Container className={classes.root}>
      <Typography align="center" variant="h3" component="h1" color="secondary">
        {user.name}{" "}
        <FormControlLabel
          control={
            <Switch
              checked={active}
              onChange={handleChange}
              name="active"
              color="primary"
            />
          }
          label={active ? "active" : "away"}
          className={classes.switch}
        />
      </Typography>
      <Grid container spacing={1}>
        <Grid item xs={12} sm={6} className={classes.item1}>
          <div>
            <Typography
              gutterBottom
              variant="h5"
              component="h2"
              color="secondary"
            >
              User Details{" "}
              <IconButton aria-label="edit user details" color="secondary">
                <EditIcon />
              </IconButton>
            </Typography>
            <Typography paragraph gutterBottom>
              <b>Name: </b>
              {user.name}
            </Typography>
            <Typography paragraph gutterBottom>
              <b>Email: </b>
              {user.email}
            </Typography>
            <Typography paragraph gutterBottom>
              <b>About: </b>
              {user.bio}
            </Typography>
            <Typography
              gutterBottom
              variant="h5"
              component="h2"
              color="secondary"
              className={classes.root}
            >
              Languages{" "}
              <IconButton aria-label="edit languages" color="secondary">
                <EditIcon />
              </IconButton>
            </Typography>
            <Typography paragraph gutterBottom>
              <b>Native Language: </b>
              {user.speaks[0].language}
            </Typography>
            <Typography gutterBottom>
              <b>Learning: </b>
              {user.learning[0].language}
            </Typography>
            <div>
              Level
              <LinearProgress
                variant="determinate"
                value={(user.learning[0].level / 4) * 100}
                className={classes.level}
              />
            </div>
          </div>
        </Grid>
        <Grid item xs={12} sm={6} className={classes.item2}>
          <div>
            <Typography
              gutterBottom
              variant="h5"
              component="h2"
              color="secondary"
            >
              Profile Picture
            </Typography>
            <ImageUploader imageUrl={user.imageUrl} />
          </div>
        </Grid>
      </Grid>
    </Container>
  );
}

export default Profile;
