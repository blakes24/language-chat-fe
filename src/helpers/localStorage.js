export function setLocalStorage(key, val) {
  window.localStorage.setItem(key, JSON.stringify(val));
}

export function getFromLocalStorage(key, defaultValue) {
  let value;
  try {
    value = JSON.parse(
      window.localStorage.getItem(key) || JSON.stringify(defaultValue)
    );
  } catch (e) {
    console.error(e);
    value = defaultValue;
  }
  return value;
}
