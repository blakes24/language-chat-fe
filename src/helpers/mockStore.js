import configureStore from "redux-mock-store";

const middlewares = [];
const mockStore = configureStore(middlewares);

const testUser = {
  id: 1,
  name: "Test",
  bio: "stuff about mae",
  email: "test@mail.com",
  imageUrl: "https://cdn.fakercloud.com/avatars/happypeter1983_128.jpg",
  active: true,
  speaks: [
    {
      code: "it",
      language: "Italian",
    },
  ],
  learning: [
    {
      code: "ko",
      language: "Korean",
      level: "2",
    },
  ],
};

const initialState = {
  users: {
    current: testUser,
    items: [],
    token: "",
    loading: "idle",
    error: null,
  },
  rooms: { items: [], current: null, loading: "idle", error: null },
  messages: { items: [], loading: "idle", error: null },
  partners: { items: [], loading: "idle", error: null },
};

export const store = mockStore(initialState);
