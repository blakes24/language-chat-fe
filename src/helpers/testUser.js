import UserContext from "./UserContext";

const testUser = {
  id: 1,
  name: "Test",
  bio: "stuff about mae",
  email: "test@mail.com",
  imageUrl: "https://cdn.fakercloud.com/avatars/happypeter1983_128.jpg",
  active: true,
  speaks: [
    {
      code: "it",
      language: "Italian",
    },
  ],
  learning: [
    {
      code: "ko",
      language: "Korean",
      level: "2",
    },
  ],
};

const UserProvider = ({
  children,
  user = testUser,
  setToken = () => console.log("set token"),
}) => <UserContext.Provider value={{ user }}>{children}</UserContext.Provider>;

export { UserProvider };
