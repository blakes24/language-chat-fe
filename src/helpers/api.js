import axios from "axios";
import { BASE_URL } from "../config";
import { getFromLocalStorage } from "./localStorage";

/** API Class: methods used to connect to the API. */

let token = getFromLocalStorage("token", "");

class ChatApi {
  static token = token;

  static async request(endpoint, data = {}, method = "get") {
    console.debug("API Call:", endpoint, data, method);
    const url = `${BASE_URL}/${endpoint}`;
    const headers = { Authorization: `Bearer ${ChatApi.token}` };
    const params = method === "get" ? data : {};
    try {
      return (await axios({ url, method, data, params, headers })).data;
    } catch (err) {
      let message = err.response
        ? err.response.data.error.message
        : "Server error: try again later";
      throw Array.isArray(message) ? message : [message];
    }
  }

  /** Create a new user and get auth token. */

  static async register(userData) {
    let res = await this.request(`auth/register`, userData, "post");
    this.token = res.token;
    return res.token;
  }

  /** Get user details. */

  static async getUser(userId) {
    let res = await this.request(`users/${userId}`);
    return res.user;
  }

  /** Verifies that the social token provided is valid */

  static async verify(token) {
    let res = await this.request(`auth/validate`, token, "post");
    return res;
  }

  /** Get gets token to authenticate future requests to api */

  static async getToken(data) {
    let res = data.email
      ? await this.request(`auth/token`, data, "post")
      : await this.request(`auth/social-token`, data, "post");
    this.token = res.token;
    return res.token;
  }

  /** Get all users, filter by speaks or learning language */

  static async getAllUsers(filters) {
    let res = await this.request(`users`, filters);
    return res.users;
  }

  /** Get a users chat rooms, filter by chat partner */

  static async getRooms({ userId, partnerId }) {
    let queryParams = { partner: partnerId };
    let res = await this.request(`users/${userId}/rooms`, queryParams);
    return res;
  }

  /** Creates a chat room for two users data = {user1, user2} */

  static async createRoom(data) {
    let res = await this.request(`rooms`, data, "post");
    return res;
  }

  /** Gets messages, filter by room */

  static async getMessages(roomId) {
    let queryParams = { room: roomId };
    let res = await this.request(`messages`, queryParams);
    return res.messages;
  }

  /** Update user info. */

  static async updateUser(id, data) {
    let res = await this.request(`users/${id}`, data, "patch");
    return res.user;
  }

  /** Update language(speaks). */

  static async updateSpeaks(id, data) {
    let res = await this.request(`languages/speaks/${id}`, data, "patch");
    return res.language;
  }

  /** Update language(learning). */

  static async updateLearning(id, data) {
    let res = await this.request(`languages/learning/${id}`, data, "patch");
    return res.language;
  }

  /** Get a users partners */

  static async getPartners(userId) {
    let res = await this.request(`users/${userId}/partners`);
    return res.partners;
  }

  /** Adds a partner */

  static async addPartner(userId, data) {
    let res = await this.request(`users/${userId}/partners`, data, "post");
    return res;
  }

  /** Deletes a partner */

  static async deletePartner({ userId, partnerId }, data) {
    let res = await this.request(
      `users/${userId}/partners/${partnerId}`,
      data,
      "delete"
    );
    return res;
  }

  /** Deletes a user */

  static async deleteUser(userId, data) {
    let res = await this.request(`users/${userId}`, data, "delete");
    return res;
  }
}

export default ChatApi;
