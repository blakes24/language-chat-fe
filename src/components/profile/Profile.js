import ImageUploader from "./ImageUploader";
import {
  Container,
  Typography,
  Grid,
  makeStyles,
  LinearProgress,
  IconButton,
  Button,
  DialogActions,
  DialogContentText,
} from "@material-ui/core";
import Dialog from "@material-ui/core/Dialog";
import DialogContent from "@material-ui/core/DialogContent";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Switch from "@material-ui/core/Switch";
import EditIcon from "@material-ui/icons/Edit";
import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { updateCurrentUser, deleteUser } from "../../store/usersSlice";
import { setLocalStorage } from "../../helpers/localStorage";
import EditUserModal from "./EditUserModal";
import EditLangModal from "./EditLangModal";
import { useHistory } from "react-router-dom";
import Snackbar from "@material-ui/core/Snackbar";
import CloseIcon from "@material-ui/icons/Close";

const useStyles = makeStyles((theme) => ({
  root: {
    marginTop: "1rem",
    maxWidth: 900,
    [theme.breakpoints.up("sm")]: {
      marginTop: "2rem",
    },
  },
  item1: {
    display: "flex",
    padding: "1rem",
    marginTop: "1rem",
    justifyContent: "center",
    order: 2,
    [theme.breakpoints.up("sm")]: {
      order: 1,
      marginTop: "2rem",
    },
  },
  item2: {
    display: "flex",
    padding: "1rem",
    marginTop: "1rem",
    justifyContent: "center",
    order: 1,
    [theme.breakpoints.up("sm")]: {
      order: 2,
      marginTop: "2rem",
    },
  },
  level: {
    width: "70%",
    marginTop: ".4rem",
    marginBottom: ".4rem",
  },
  switch: {
    marginLeft: ".5rem",
  },
  btn: {
    marginTop: "2rem",
  },
}));

function Profile() {
  const user = useSelector((state) => state.users.current);
  const loading = useSelector((state) => state.users.loading);
  const error = useSelector((state) => state.users.error);
  const classes = useStyles();
  const [langOpen, setLangOpen] = useState(false);
  const [userOpen, setUserOpen] = useState(false);
  const [deleteOpen, setDeleteOpen] = useState(false);
  const [toastOpen, setToastOpen] = useState(false);
  const [langType, setLangType] = useState("speaks");
  const dispatch = useDispatch();
  const history = useHistory();

  const handleSpeaksOpen = () => {
    setLangType("speaks");
    setLangOpen(true);
  };
  const handleLearningOpen = () => {
    setLangType("learning");
    setLangOpen(true);
  };
  const handleUserOpen = () => {
    setUserOpen(true);
  };
  const handleLangClose = () => {
    setLangOpen(false);
  };
  const handleUserClose = () => {
    setUserOpen(false);
  };
  const handleChange = () => {
    dispatch(updateCurrentUser({ active: !user.active, id: user.id }));
  };

  const deleteAccount = () => {
    dispatch(deleteUser(user.id));
    if (loading === "idle" && !error) {
      setLocalStorage("token", "");
      history.push("/");
    } else {
      setToastOpen(true);
      setDeleteOpen(false);
    }
  };

  return (
    <Container className={classes.root}>
      <Typography align="center" variant="h3" component="h1" color="secondary">
        {user.name}{" "}
        <FormControlLabel
          control={
            <Switch
              checked={user.active}
              onChange={handleChange}
              name="active"
              color="primary"
            />
          }
          label={user.active ? "active" : "away"}
          className={classes.switch}
        />
      </Typography>
      <Grid container spacing={1}>
        <Grid item xs={12} sm={6} className={classes.item1}>
          <div>
            <Typography
              gutterBottom
              variant="h5"
              component="h2"
              color="secondary"
            >
              User Details{" "}
              <IconButton
                aria-label="edit user details"
                color="secondary"
                onClick={handleUserOpen}
              >
                <EditIcon />
              </IconButton>
            </Typography>
            <Typography paragraph gutterBottom>
              <b>Name: </b>
              {user.name}
            </Typography>
            <Typography paragraph gutterBottom>
              <b>Email: </b>
              {user.email}
            </Typography>
            <Typography paragraph gutterBottom>
              <b>About: </b>
              {user.bio}
            </Typography>
            <Typography
              gutterBottom
              variant="h5"
              component="h2"
              color="secondary"
              className={classes.root}
            >
              Languages{" "}
            </Typography>
            <Typography>
              <b>Native Language: </b>
              {user.speaks[0].language}
              <IconButton
                aria-label="edit native language"
                color="secondary"
                onClick={handleSpeaksOpen}
              >
                <EditIcon />
              </IconButton>
            </Typography>
            <Typography gutterBottom>
              <b>Learning: </b>
              {user.learning[0].language}
              <IconButton
                aria-label="edit learning language"
                color="secondary"
                onClick={handleLearningOpen}
              >
                <EditIcon />
              </IconButton>
            </Typography>
            <div>
              Level
              <LinearProgress
                variant="determinate"
                value={(user.learning[0].level / 4) * 100}
                className={classes.level}
              />
            </div>
            <Button
              color="primary"
              onClick={() => setDeleteOpen(true)}
              variant="contained"
              className={classes.btn}
            >
              Delete Account
            </Button>
          </div>
        </Grid>
        <Grid item xs={12} sm={6} className={classes.item2}>
          <div>
            <Typography
              gutterBottom
              variant="h5"
              component="h2"
              color="secondary"
            >
              Profile Picture
            </Typography>
            <ImageUploader imageUrl={user.imageUrl} userId={user.id} />
          </div>
        </Grid>
      </Grid>
      <EditUserModal open={userOpen} handleClose={handleUserClose} />
      <EditLangModal
        open={langOpen}
        type={langType}
        handleClose={handleLangClose}
      />
      <Dialog
        open={deleteOpen}
        onClose={() => setDeleteOpen(false)}
        aria-labelledby="delete-alert"
      >
        <DialogContent>
          <DialogContentText id="delete-alert" color="secondary">
            Are you sure you want to delete your account?
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button
            onClick={() => setDeleteOpen(false)}
            variant="contained"
            color="secondary"
            id="cancel"
          >
            Cancel
          </Button>
          <Button
       
                id="delete"
  
                     onClick={deleteAccount}
     
                  color="primary"
        
               variant="contained"
          
          >
            Delete
          </Button>
        </DialogActions>
      </Dialog>
      <Snackbar
        anchorOrigin={{
          vertical: "bottom",
          horizontal: "left",
        }}
        open={toastOpen}
        autoHideDuration={5000}
        onClose={() => setToastOpen(false)}
        message={error}
        action={
          <IconButton
            size="small"
            aria-label="close"
            color="inherit"
            onClick={() => setToastOpen(false)}
          >
            <CloseIcon fontSize="small" />
          </IconButton>
        }
      />
    </Container>
  );
}

export default Profile;
