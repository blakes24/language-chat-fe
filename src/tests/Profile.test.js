import { render } from "@testing-library/react";
import Profile from "../components/profile/Profile";
import { MemoryRouter } from "react-router-dom";
import { store } from "../helpers/mockStore";
import { Provider } from "react-redux";

test("renders without crashing", () => {
  render(
    <Provider store={store}>
      <MemoryRouter>
        <Profile />
      </MemoryRouter>
    </Provider>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Profile />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
