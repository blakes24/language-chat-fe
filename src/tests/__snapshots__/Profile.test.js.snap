// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-root-23 MuiContainer-maxWidthLg"
  >
    <h1
      class="MuiTypography-root MuiTypography-h3 MuiTypography-colorSecondary MuiTypography-alignCenter"
    >
      Test 
      <label
        class="MuiFormControlLabel-root makeStyles-switch-27"
      >
        <span
          class="MuiSwitch-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-checked-30 Mui-checked"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-32 MuiSwitch-input"
                name="active"
                type="checkbox"
                value=""
              />
              <span
                class="MuiSwitch-thumb"
              />
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          active
        </span>
      </label>
    </h1>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
    >
      <div
        class="MuiGrid-root makeStyles-item1-24 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div>
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            User Details 
            <button
              aria-label="edit user details"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiTypography-paragraph"
          >
            <b>
              Name: 
            </b>
            Test
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiTypography-paragraph"
          >
            <b>
              Email: 
            </b>
            test@mail.com
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiTypography-paragraph"
          >
            <b>
              About: 
            </b>
            stuff about mae
          </p>
          <h2
            class="MuiTypography-root makeStyles-root-23 MuiTypography-h5 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            Languages 
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <b>
              Native Language: 
            </b>
            Italian
            <button
              aria-label="edit native language"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            <b>
              Learning: 
            </b>
            Korean
            <button
              aria-label="edit learning language"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </p>
          <div>
            Level
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="50"
              class="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-level-26 MuiLinearProgress-determinate"
              role="progressbar"
            >
              <div
                class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
                style="transform: translateX(-50%);"
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-28 MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Delete Account
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-item2-25 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div>
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            Profile Picture
          </h2>
          <div
            class="makeStyles-root-33"
          >
            <div
              class="makeStyles-imgContainer-35"
            >
              <img
                alt="avatar"
                class="makeStyles-img-36"
                src="https://cdn.fakercloud.com/avatars/happypeter1983_128.jpg"
              />
            </div>
            <form
              class="makeStyles-form-34"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-fullWidth"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Change Image
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <input
                accept="image/*"
                style="display: none;"
                type="file"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
